library/std/src/primitive_docs.rs

```diff
@@ -445,7 +445,26 @@ mod prim_unit {}
 /// Note that here the call to [`drop`] is for clarity - it indicates
 /// that we are done with the given value and it should be destroyed.
 ///
-/// ## 3. Get it from C.
+/// ## 3. Create it using `ptr::addr_of!`
+///
+/// Instead of coercing a reference to a raw pointer, you can use the macros
+/// [`ptr::addr_of!`] (for `*const T`) and [`ptr::addr_of_mut!`] (for `*mut T`).
+/// Those macros have an advantage over the coercion: you can create raw
+/// pointers to unaligned fields, which might be necessary if packed structs
+/// are involved.
```

Comment:
This makes it sound like unaligned fields are the only reason to use addr_of instead of coercing from a reference. But making pointers to e.g. uninitialized data is another important use case.